Redefinition error building driver header in c++ code

<!-- Please use this template while reporting a bug and provide as much info as possible. Not doing so may result in your bug not being addressed in a timely manner. Thanks! -->

**Describe the bug**

When including in C++ code, `ppm_events_public.h` header causes the following error:
```
In file included from /home/user/workdir/falco/userspace/libscap/scap.h:64,
                 from /home/user/workdir/falco/userspace/libsinsp/sinsp.h:69,
                 from /home/user/workdir/code.cpp:31:
/home/user/workdir/falco/userspace/libscap/../../driver/ppm_events_public.h:1802:18: error: declaration of ‘int (* ppm_event_entry::filler_callback)(event_filler_arguments*)’ changes meaning of ‘filler_callback’ [-fpermissive]
 1802 |  filler_callback filler_callback;
      |                  ^~~~~~~~~~~~~~~
/home/user/workdir/falco/userspace/libscap/../../driver/ppm_events_public.h:1799:15: note: ‘filler_callback’ declared here as ‘typedef int (* filler_callback)(struct event_filler_arguments*)’
 1799 | typedef int (*filler_callback) (struct event_filler_arguments *args);
```

**How to reproduce it**

<!-- Minimal and precise steps to reproduce the bug. -->
Compile the following as C++ with `gcc`
```
#include "falcosecurity-libs/driver/ppm_events_public.h"

int main() {
    return 0;
}
```

**Expected behaviour**

<!-- A clear and concise description of what you expected to happen. -->
Compilation succeeds

**Environment**

<!-- Please complete the following info. -->

- OS:
Ubuntu 20.04.4 LTS (Focal Fossa)

- Kernel:
5.13.0-51-generic

- GCC:
```
gcc -v
Using built-in specs.
COLLECT_GCC=gcc
COLLECT_LTO_WRAPPER=/usr/lib/gcc/x86_64-linux-gnu/9/lto-wrapper
OFFLOAD_TARGET_NAMES=nvptx-none:hsa
OFFLOAD_TARGET_DEFAULT=1
Target: x86_64-linux-gnu
Configured with: ../src/configure -v --with-pkgversion='Ubuntu 9.4.0-1ubuntu1~20.04.1' --with-bugurl=file:///usr/share/doc/gcc-9/README.Bugs --enable-languages=c,ada,c++,go,brig,d,fortran,objc,obj-c++,gm2 --prefix=/usr --with-gcc-major-version-only --program-suffix=-9 --program-prefix=x86_64-linux-gnu- --enable-shared --enable-linker-build-id --libexecdir=/usr/lib --without-included-gettext --enable-threads=posix --libdir=/usr/lib --enable-nls --enable-clocale=gnu --enable-libstdcxx-debug --enable-libstdcxx-time=yes --with-default-libstdcxx-abi=new --enable-gnu-unique-object --disable-vtable-verify --enable-plugin --enable-default-pie --with-system-zlib --with-target-system-zlib=auto --enable-objc-gc=auto --enable-multiarch --disable-werror --with-arch-32=i686 --with-abi=m64 --with-multilib-list=m32,m64,mx32 --enable-multilib --with-tune=generic --enable-offload-targets=nvptx-none=/build/gcc-9-Av3uEd/gcc-9-9.4.0/debian/tmp-nvptx/usr,hsa --without-cuda-driver --enable-checking=release --build=x86_64-linux-gnu --host=x86_64-linux-gnu --target=x86_64-linux-gnu
Thread model: posix
gcc version 9.4.0 (Ubuntu 9.4.0-1ubuntu1~20.04.1)
```

--- a/driver/ppm_events_public.h
+++ b/driver/ppm_events_public.h
@@ -1802,10 +1802,10 @@ enum autofill_paramtype {
 	APT_SOCK,
 };
 
-typedef int (*filler_callback) (struct event_filler_arguments *args);
+typedef int (*filler_callback_t) (struct event_filler_arguments *args);
 
 struct ppm_event_entry {
-	filler_callback filler_callback;
+	filler_callback_t filler_callback;
 	enum ppm_filler_id filler_id;
 	uint16_t n_autofill_args;
 	enum autofill_paramtype paramtype;

